discovery/gossiper_test.go

```diff
@@ -170,8 +166,25 @@ func (r *mockGraphSource) ForEachNode(func(node *channeldb.LightningNode) error)
 	return nil
 }
 
-func (r *mockGraphSource) ForAllOutgoingChannels(cb func(i *channeldb.ChannelEdgeInfo,
-	c *channeldb.ChannelEdgePolicy) error) error {
+func (r *mockGraphSource) ForAllOutgoingChannels(
+	cb func(i *channeldb.ChannelEdgeInfo,
+		c *channeldb.ChannelEdgePolicy) error) error {
+
+	for _, info := range r.infos {
+		e, ok := r.edges[info.ChannelID]
+		if !ok {
+			return fmt.Errorf("policy for chanID %v not found",
+				info.ChannelID)
+		}
+		if len(e) != 2 {
+			return fmt.Errorf("had %d policies for chanID %v",
+				len(e), info.ChannelID)
+		}
+		if err := cb(info, e[0]); err != nil {
```

Comment:
Doesn't seem like we can guarantee that the first edge will always be the outgoing one?

Reply:
Fixed!